### PR TITLE
Feature/Rating

### PR DESCRIPTION
**Pull request by lschumac24**
In der Details Activity wird nun der Durchschnitt der eigenen Bewertungen angezeigt.
Eine neue Bewertung kann mit Aktionien->Bewerten erstellt werden. #

### CODE REVIEW

File: app/src/main/java/ch/beerpro/presentation/details/DetailsViewModel.java [hunk 1]
```diff
@@ -64,4 +68,14 @@ public void toggleLike(Rating rating) {
...
     }
+
+    public float calcAvgRating(List<Rating> ratings){
+        float sum = 0;
+        if(!ratings.isEmpty()){
+            for(Rating rating : ratings){
+                sum += rating.getRating();
+            }
+        }
+        return sum/ratings.size();
```
Comment (rkram3r) [inline app/src/main/java/ch/beerpro/presentation/details/DetailsViewModel.java]: Is calcAvgRating always called when ratings are changed?
If yes, possible divided by 0 exception:
1. Add rating (size = 1)
2. Remove rating (size = 0)

File: app/src/main/java/ch/beerpro/presentation/details/DetailsViewModel.java [hunk 2]
```diff
@@ -53,6 +55,8 @@ public DetailsViewModel() {
         return ratings;
     }
 
+    public LiveData<List<Rating>> getMyRatings() {return myRatings;}
```
Comment (rkram3r) [inline app/src/main/java/ch/beerpro/presentation/details/DetailsViewModel.java]: Reformat file.

File: app/src/main/java/ch/beerpro/presentation/details/DetailsActivity.java [hunk 3]
```diff
@@ -125,6 +122,13 @@ public void showBottomSheetDialog() {
         View view = getLayoutInflater().inflate(R.layout.single_bottom_sheet_dialog, null);
         BottomSheetDialog dialog = new BottomSheetDialog(this);
         dialog.setContentView(view);
+
+        Button ratingButton = (Button)dialog.findViewById(R.id.giveRating);
+        ratingButton.setOnClickListener(v -> {
```
Comment (rkram3r) [inline app/src/main/java/ch/beerpro/presentation/details/DetailsActivity.java]: I would write out the variable names to improve legibility.

File: app/src/main/java/ch/beerpro/presentation/details/DetailsActivity.java [hunk 4]
```diff
@@ -125,6 +122,13 @@ public void showBottomSheetDialog() {
...
+            addNewRating(myRatingBar);
+            dialog.dismiss();
+        });
```
Comment (rkram3r) [inline app/src/main/java/ch/beerpro/presentation/details/DetailsActivity.java:130]: I would write out the variable names to improve legibility.

**Review by rkram3r — COMMENTED** [review]
Good.  👍 
Im not sure if calcAvgRating contains a small bug.

File: app/src/main/java/ch/beerpro/presentation/details/DetailsActivity.java (hunk 4 above)
Comment (lschumac24) [reply app/src/main/java/ch/beerpro/presentation/details/DetailsActivity.java:130]: This is how android automatically creates Lambdas

**Review by rkram3r — APPROVED** [review]
👍

**Pull request merged**